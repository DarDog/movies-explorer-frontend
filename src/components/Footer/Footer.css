.footer {
  background-color: #FAFAFA;
  padding: 80px 70px 20px;
  width: 100%;
  max-width: 1280px;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
}

.footer__paragraph {
  margin: 0 0 20px;
  font-weight: 400;
  font-size: 13px;
  line-height: 16px;
  text-align: center;
  color: #A0A0A0;
  width: 100%;
  border-bottom: 1px solid #E8E8E8;
  padding-bottom: 20px;
}

.footer__info {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.footer__copyright {
  font-weight: 400;
  font-size: 13px;
  line-height: 16px;
  color: black;
  margin: 0;
}

.footer__list {
  gap: 20px;
}

.footer__item {
  transition: .3s;
}

.footer__item:hover {
  opacity: .5;
}

.footer__link {
  text-decoration: none;
  margin: 0;
  font-weight: 400;
  font-size: 13px;
  line-height: 16px;
  text-align: right;
  color: black;
}


@media screen and (max-width: 800px){
  .footer {
    padding: 80px 30px 20px;
    background-color: white;
  }
}

@media screen and (max-width: 550px){
  .footer {
    padding: 80px 10px 10px;
  }

  .footer__paragraph {
    margin-bottom: 30px;
    font-size: 12px;
    line-height: 15px;
  }

  .footer__info {
    flex-direction: column-reverse;
    gap: 30px;
  }

  .footer__list {
    flex-direction: column;
    gap: 12px;
    align-items: center;
  }
}
