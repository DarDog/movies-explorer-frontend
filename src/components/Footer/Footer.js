import React from 'react';
import './Footer.css'
import Ul from "../Ul/Ul";

const Footer = () => {
  return(
    <footer className='footer'>
      <p className='footer__paragraph'>Учебный проект Яндекс.Практикум х BeatFilm.</p>
      <div className='footer__info'>
        <p className="footer__copyright">&#169; 2022</p>
        <Ul anotherClass='footer__list'>
          <li className='footer__item'><a href="https://practicum.yandex.ru/" className='footer__link'>Яндекс.Практикум</a></li>
          <li className='footer__item'><a href="https://github.com/DarDog" className='footer__link'>Github</a></li>
          <li className='footer__item'><a href="https://vk.com/me_here_fine" className='footer__link'>VK</a></li>
        </Ul>
      </div>
    </footer>
  );
}

export default Footer
