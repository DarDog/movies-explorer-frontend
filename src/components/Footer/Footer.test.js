import React from 'react';
import Footer from "./Footer";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';


describe('Компонент Ul', () => {
  it('Рендерится', () => {
    render(<Footer />);

    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('listitem')).toHaveLength(3)
  })
  it('Соответствует своему snapshot', () => {
    const component = render(<Footer />);

    expect(component).toMatchSnapshot();
  })
})
