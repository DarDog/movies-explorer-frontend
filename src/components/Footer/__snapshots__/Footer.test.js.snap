// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Ul Соответствует своему snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
      >
        <p
          class="footer__paragraph"
        >
          Учебный проект Яндекс.Практикум х BeatFilm.
        </p>
        <div
          class="footer__info"
        >
          <p
            class="footer__copyright"
          >
            © 2022
          </p>
          <ul
            class="list footer__list "
          >
            <li
              class="footer__item"
            >
              <a
                class="footer__link"
                href="https://practicum.yandex.ru/"
              >
                Яндекс.Практикум
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="footer__link"
                href="https://github.com/DarDog"
              >
                Github
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="footer__link"
                href="https://vk.com/me_here_fine"
              >
                VK
              </a>
            </li>
          </ul>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
    >
      <p
        class="footer__paragraph"
      >
        Учебный проект Яндекс.Практикум х BeatFilm.
      </p>
      <div
        class="footer__info"
      >
        <p
          class="footer__copyright"
        >
          © 2022
        </p>
        <ul
          class="list footer__list "
        >
          <li
            class="footer__item"
          >
            <a
              class="footer__link"
              href="https://practicum.yandex.ru/"
            >
              Яндекс.Практикум
            </a>
          </li>
          <li
            class="footer__item"
          >
            <a
              class="footer__link"
              href="https://github.com/DarDog"
            >
              Github
            </a>
          </li>
          <li
            class="footer__item"
          >
            <a
              class="footer__link"
              href="https://vk.com/me_here_fine"
            >
              VK
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
